src/client/caching/structures/guilds/Guild.ts

```diff
@@ -0,0 +1,427 @@
+	 */
+	public readonly voiceStates: VoiceStateStore;
+
+	/**
+	 * A store of members for this guild.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/guild#guild-member-object
+	 */
+	public readonly members: GuildMemberStore;
+
+	/**
+	 * A store of channels for this guild.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/channel#channel-object
+	 */
+	public readonly channels: GuildChannelStore;
+
+	/**
+	 * A store of presences for this guild.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/topics/gateway#presence-update
+	 */
+	public readonly presences: PresenceStore;
+
+	/**
+	 * The maximum amount of presences for the guild (the default value, currently 25000, is in effect when null is returned).
+	 * @since 0.0.1
+	 */
+	public maxPresences?: number | null;
+
+	/**
+	 * The maximum amount of members for the guild.
+	 * @since 0.0.1
+	 */
+	public maxMembers?: number;
+
+	/**
+	 * The vanity invite code for the guild.
+	 * @since 0.0.1
+	 */
+	public vanityUrlCode!: string | null;
+
+	/**
+	 * The guild's description.
+	 * @since 0.0.1
+	 */
+	public description!: string | null;
+
+	/**
+	 * The guild's banner hash.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/reference#image-formatting
+	 */
+	public banner!: string | null;
+
+	/**
+	 * The guild's Server Boosting tier
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/resources/guild#guild-object-premium-tier
+	 */
+	public premiumTier!: GuildPremiumTier | null;
+
+	/**
+	 * The number of boosts this guild currently has.
+	 * @since 0.0.1
+	 */
+	public premiumSubscriptionCount!: number | null;
+
+	/**
+	 * The preferred locale of a `PUBLIC` guild used in server discovery and notices from Discord; defaults to "en-US".
+	 * @since 0.0.1
+	 */
+	public preferredLocale!: string;
+
+	/**
+	 * The id of the channel where admins and moderators of "PUBLIC" guilds receive notices from Discord.
+	 * @since 0.0.1
+	 */
+	public publicUpdatesChannel!: string | null;
+
+	/**
+	 * The approximate number of members in this guild, returned from the `GET /guild/<id>` endpoint when `with_counts` is `true`.
+	 * @since 0.0.1
+	 */
+	public approximateMemberCount?: number;
+
+	/**
+	 * The approximate number of online members in this guild, returned from the `GET /guild/<id>` endpoint when `with_counts` is `true`.
+	 */
+	public approximatePresenceCount?: number;
+
+	public constructor(client: Client, data: APIGuildData) {
+		super(client);
+
+		this.id = data.id;
+		this.bans = new BanStore(client, this);
+		this.roles = new RoleStore(client);
+		this.emojis = new GuildEmojiStore(client);
+		this.voiceStates = new VoiceStateStore(client);
+		this.members = new GuildMemberStore(client, this);
+		this.channels = new GuildChannelStore(client);
+		this.presences = new PresenceStore(client);
+
+		if (data.unavailable) {
+			this.available = false;
```

Comment:
q: Should we stick to discord's data or keep this as is?

Reply:
@bdistin ^